### PR TITLE
Use consistent enable property for all auto config modules

### PR DESCRIPTION
**Pull request by warrenzhu25**

### CODE REVIEW

File: spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java [hunk 1]
```diff
@@ -30,14 +26,18 @@
  * @author Warren Zhu
  */
 @Configuration
-@EnableConfigurationProperties({AzureProperties.class, TelemetryProperties.class})
-@ConditionalOnClass(name = "com.microsoft.azure.management.Azure")
-@ConditionalOnProperty("spring.cloud.azure.credentialFilePath")
+@EnableConfigurationProperties(AzureProperties.class)
+@ConditionalOnClass(Azure.class)
+@ConditionalOnProperty("spring.cloud.azure.enabled")
```
Comment (Incarnation-p-lee) [inline spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java]: context should be always enabled ?

Comment (warrenzhu25) [reply spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java]: Any auto config could be disabled. And user just provide their bean to continue use other dependent parts

Comment (Incarnation-p-lee) [reply spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java]: By default, we should enable all modules, right ?

File: spring-cloud-azure-autoconfigure/src/test/java/com/microsoft/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfigurationTest.java [hunk 2]
```diff
@@ -25,23 +26,27 @@
 
     @Test
     public void testAzurePropertiesConfigured() {
-        this.contextRunner.withPropertyValues("spring.cloud.azure.credentialFilePath=credential")
+        this.contextRunner.
+                                  withPropertyValues("spring.cloud.azure.credentialFilePath=credential")
```
Comment (Incarnation-p-lee) [inline spring-cloud-azure-autoconfigure/src/test/java/com/microsoft/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfigurationTest.java]: should be aligned ?

Comment (warrenzhu25) [reply spring-cloud-azure-autoconfigure/src/test/java/com/microsoft/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfigurationTest.java]: It seems auto formatted by IntelliJ

**Pull request merged**